app/javascript/controllers/video_player_controller.js

```diff
@@ -61,4 +64,24 @@ export default class extends Controller {
       this.player.seekTo(time)
     }
   }
+
+  playOnYoutube()
+  {
+    const videoId = this.playerTarget.dataset.youtubeId
+    window.open(`https://www.youtube.com/watch?v=${videoId}`, '_blank')
+    this.player.pause()
+  }
+
+  createYouTubeIcon () {
+    const wrappedAnchorElement = document.createElement('a')
+    wrappedAnchorElement.dataset.action = "click->video-player#playOnYoutube"
```

Comment:
If you just put the `href` on the `<a>` tag itself it such more predictable and also easier code-wise imo:

```suggestion
  pause() {
    this.player.pause()
  }

  createYouTubeIcon () {
    const videoId = this.playerTarget.dataset.youtubeId
    const wrappedAnchorElement = document.createElement('a')
    wrappedAnchorElement.dataset.action = "click->video-player#playOnYoutube"
    wrappedAnchorElement.href = `https://www.youtube.com/watch?v=${videoId}`
    wrappedAnchorElement.target = "_blank"
```